concepts/00001-enarx-sprint-framework/README.md

```diff
@@ -0,0 +1,158 @@
+- Automation opportunity: we could automatically move issues from Tracking to Active when they are assigned to someone. Automatable with Actions.
+
+### Active Sprint Board
+This board is where the team will track progress on current sprint tasks. **It should hold no backlog;** if an issue or PR is on this board, it is being actively worked on.
+
+Here’s how the columns should look:
+
+- Assigned: this indicates a task that is currently being worked on as part of the sprint. Newly-added issues land here by default (automatable with stock Github).
+- Ready for review: where pull requests land by default (automatable with stock Github); indicates it’s ready to be reviewed by someone. Pull requests should move back here from Changes Requested if a fix has been applied and it’s ready for a re-review (not in stock Github, but automatable with Actions).
+- Changes Requested: PRs move here when a review is submitted that requests changes (automatable with stock Github).
+- Approved: PRs that have been approved go here, pending merge (automatable with stock Github).
+- Done: PRs that have been merged, and their associated issues, go here (automatable with stock Github).
+
+A few things to consider:
+- Advantage: having project board be separate from backlog means we can promote issues to be part of the current sprint whenever we want to, and they can skip the backlog entirely if that’s desired.
+- Automation opportunity: If pull requests have an associated issue, don’t put the PR on the board -- instead, move the associated issue to the board (if not on it already) and treat the issue like the pull request, subject to the same movement rules as specified above. This would prevent us from having to look at the same thing twice, while still preserving a direct link to the PR in the context of its issue (since Github now displays linked PRs as a dropdown on an issue).
+
+### Sprint Planning
+With our backlog and sprint boards established, we can flesh out our first meeting: sprint planning.
+
+This is meant to be a time to consider what work the team wants to accomplish in the next two-week sprint and assign relevant tasks to people. It should happen on Mondays and take about an hour, following the sprint retrospective (detailed later).
+
+It should start with a brief discussion about the team’s goals. This should be an open, but high-level, discussion that will help frame the kinds of issues we should prioritize. It shouldn’t take long -- no more than 15-20 minutes.
+
+With the context of that discussion in mind, we’ll move to backlog review. As detailed above, our backlogs are contained in individual boards, and we’ll run through each one as part of the planning process.
+
+For each board:
+- Briefly run through the Backlog column to decide if there are any issues we want to promote to "Tracking" status.
+- Run through all items in "Tracking" and decide which ones are highest priority, and who will work on them. They will become the issue’s assignee.
```

Comment:
I think it's important that we come up with some agreeable "unit" of work and estimate how many units of work comprise a "full" sprint for an engineer.

Furthermore, when grooming or picking tasks we should have some consensus based on the estimated units of work that issue will take. Just to be clear, this is not a do-or-die deadline (and it's most certainly not to be used _against_ anyone during retro), but this is a tool for helping us not overload someone with work. Estimating scope of most non-trivial issues is exceptionally hard, especially since we have such a component-based group, but I think it's still worthwhile. These metrics may be useful during the retrospective. If the unit is agreed on as a team, that ties in to the fact that we sprint as a team and we ultimately move code as a team. If issues are left unfinished due to scope or difficulty, then at least we can share that learning experience as a team.

Reply:
I agree, I think what a "unit" of work is and what each task is worth should be considered carefully. I will admit up front that I have no idea how long a task might take more than half the time, because I mostly do things I've never done before. Sometimes a task I think will take a day takes three weeks, and sometimes (rarely) a task I think will take two weeks takes an afternoon. Others can probably help out here in estimating, but we also have to keep in mind that something that might take a senior engineer two days could take a more junior engineer two weeks or an intern a month. IMO this is the biggest challenge with moving to a "sprint" structure since so much of our project is uncharted territory and new problems come up en route to even straightforward-seeming issues.

Reply:
There are also particular roles to be considered here: it's not just a free-for-all discussion.  It needs to be led and coordinated.

Reply:
The unit of work is supposed to stem from the "complexity points", AFAIK. See comment above regarding backlog grooming vs. sprint planning on this.

Reply:
I think work estimation is indeed important. However, I think it is difficult and I'd like for us to ease our way into this process. What do people think about leaving the unit undefined for now so that we can test our workflow and automation? Then we can add in "work units" and more careful planning as a follow-up.

I'm thinking perhaps we outline the workflow first and then implement it. This puts the RFC in the `demonstrated` state. Then we add "work units" before the transition to `adopted`. Thoughts?

Reply:
I like the idea of leaving the "work unit" undefined at first and getting the rest of the process working, since I think estimating work will be a challenge in itself.